pkg/proxy/accesslog/log.go

```diff
@@ -129,8 +134,80 @@ func Log(l *LogRecord, typ FlowType, verdict FlowVerdict, code int) {
 	} else {
 		logString(string(b), true)
 	}
+}
+
+// LogKafka logs a Kafka record to the logfile
+func LogKafka(l *LogRecord, typ FlowType, verdict FlowVerdict, code int) {
+	apiKey := l.KafkaRequest.GetAPIKey()
+	if apiKey < 0 || apiKey > api.KafkaMaxAPIKeyVal {
+		log.WithFields(log.Fields{
+			FieldFilePath: logPath,
+		}).Error("Invalid Kafka Api Key")
```

Comment:
Why are we not logging this? If it was observed on the wire then we should log it in the access log. I understand that this is protecting the map access. What I would do here instead is create a function which checks whether there is a hit in the map and if no hit is found, encode the apiKey as number in string representation.

Reply:
I personally think that since this has already gone through canAccess by the time we actually log it, the apiKey is already checked for correct values, so this failure will not be hit Imo, please correct me if I am wrong. Let me re-organise and see how I can change the code here.

Reply:
I see that it can be hit before `canAccess`. Ignore my earlier comment

Reply:
changed as per discussion offline.